packages/base-controller/src/ControllerMessenger.ts

```diff
@@ -53,9 +53,11 @@ export type ExtractEventPayload<
 
 export type GenericEventHandler = (...args: unknown[]) => void;
 
-export type SelectorFunction<Event extends EventConstraint, ReturnValue> = (
-  ...args: ExtractEventPayload<Event>
-) => ReturnValue;
+export type SelectorFunction<
+  Event extends EventConstraint,
+  ReturnValue,
+  EventType = Event['type'],
```

Comment:
Why would a third type parameter be needed here? The type is part of the event, which is the first parameter.

It's unclear how this relates to the other changes in this PR